src/conjecture_tactics.lean

```diff
@@ -0,0 +1,93 @@
+
+import tactic
+import tactic.gptf.basic
+import util.tactic
+
+namespace tactic.interactive
+/--
+  returns old tactic state, narrowed tactic state
+-/
+meta def tac₁ (nm_str : string) (conj_str : string) : tactic (tactic_state × tactic_state) := do {
+  ts_old ← tactic.read,
+  let tac_str := format! "have {nm_str} : {conj_str}",
+  result ← get_tac_and_capture_result tac_str.to_string 5000,
+  return_result ← (match result with
+  | interaction_monad.result.success _ ts' := do {
+  ts_narrowed ← do {
+    tactic.write ts',
+    g ← list.head <$> tactic.get_goals,
+    tactic.set_goals [g],
+    tactic.read
+  },
+  pure $ prod.mk ts' ts_narrowed
+}
+  | interaction_monad.result.exception fn pos ts' := do {
+    let thunk := fn.get_or_else (λ _, format! "exception"),
+    tactic.fail format! "{thunk ()}"
+  }
+  end),
+
+  tactic.write ts_old,
+  pure return_result
+} 
+
+-- Generates names for conjectures given a decl name
+meta def name_conj (old : name) (new : option name := none) : tactic name :=
+  match new with
+  | none :=
+    match old.components.reverse with
+    | last :: most := (do let last := last.to_string,
+                         tactic.get_unused_name $ mk_str_name old.get_prefix last ++ "_conjecture")
+    | nil          := undefined
+    end
+  | (some new) := return (mk_str_name old.get_prefix new.to_string)
+end
+
+
+/--  
+  return value ⟨nm, ts⟩ satisfies that `nm` is the name of a new declaration whose value is the proof `conj_pf` of the conjecture and that `ts` contains an environment which has `nm` registered as a declaration.
+-/
+meta def tac₂ (conj_pf : expr) (narrowed_ts : tactic_state) : tactic (name × tactic_state) := do {
```

Comment:
@jesse-michael-han how does this second tactic look?

Reply:
tentatively LGTM, can you try testing it with something like this?

```lean
/--
  should succeed iff tac₂ meets its spec
-/
meta def test_tac₂ : tactic unit := do {
  let pf_term : expr := `(trivial),
  ts ← tactic.read,
  ⟨nm ,new_ts⟩ ← tac₂ pf_term ts,
  tactic.write new_ts,
  env ← tactic.get_env,
  decl ← env.get nm, -- succeeds iff `nm` is inside `env`
  pure ()
}

```